src/multibody/model.hxx

```diff
@@ -251,16 +251,21 @@ namespace pinocchio
   inline int ModelTpl<Scalar,Options,JointCollectionTpl>::
   addFrame(const Frame & frame)
   {
-    if(!existFrame(frame.name, frame.type))
+    // Check if the frame.name exists
+    if(existFrame(frame.name))
+    {
+      FrameIndex frame_id = getFrameId(frame.name);
+      if(frames[frame_id].type == frame.type)
```

Comment:
Is there any reason to check that the frame type matches and not the other values (placement, parent joint and parent frame, etc) ?

Reply:
Maybe not. I can change it if you think we need to encompass a broader range of behaviors.
I think the type is only informative, no more.

Reply:
I guess we need to check `frame.placement` and `frame.previousFrame`. If one of this two is different, then probably we aren't doing something correct...

Reply:
So calling Frame::operator= should do the job then.